probe/host/system_linux.go

```diff
@@ -31,6 +33,67 @@ var GetKernelReleaseAndVersion = func() (string, string, error) {
 	return string(release), string(version), nil
 }
 
+// GetPrettyName extracts the PRETTY_NAME from host's /etc/os-release
+// SPEC: https://www.freedesktop.org/software/systemd/man/os-release.html
+var GetPrettyName = func () (string, error) {
+
+	var buffer bytes.Buffer
+
+	buf, err := ioutil.ReadFile("/var/run/scope/host-os-release")
+	if err != nil {
+		return runtime.GOOS, nil
+	}
+
+	// If can't find host-os-release, get container's os-release
+	if buf == nil {
+		bufContainer, err := ioutil.ReadFile("/etc/os-release")
+		if err != nil {
+			return runtime.GOOS, nil
+		}
+		buf = bufContainer
+	}
+
+	// Pretty Name Fallback Flow:
+	// 1. $PRETTY_NAME
+	// 2. $NAME $VERSION
+	// 3. $ID $VERSION_ID
+	// 4. runtime.GOOS
+
+	prettyNameParse := regexp.MustCompile("PRETTY_NAME=(.+?)\n")
+	nameParse := regexp.MustCompile("NAME=(.+?)\n")
+	versionParse := regexp.MustCompile("VERSION=(.+?)\n")
+	IDParse := regexp.MustCompile("ID=(.+?)\n")
+	versionIDParse := regexp.MustCompile("VERSION_ID=(.+?)\n")
+
+	prettyName := prettyNameParse.FindStringSubmatch(string(buf))
```

Comment:
Instead of using a lot of regex to parse this file (several times!) I'd cycle though the lines and use simple string functions to extract the interesting values.